py/dml/types.py

```diff
@@ -168,6 +171,27 @@ def conv_const(const, t):
         t.const = True
     return t
 
+def safe_realtype_unconst(t0):
+    def sub(t):
+        if isinstance(t, (TArray, TVector)):
+            base = sub(t.base)
+            if t.const or base is not t.base:
+                t = t.clone()
+                t.const = False
+                t.base = base
+        elif t.const:
+            t = t.clone()
+            t.const = False
+        return t
+    return sub(safe_realtype(t0))
+
+def shallow_const(t):
+    t = safe_realtype_shallow(t)
+    while not t.const and isinstance(t, (TArray, TVector)):
```

Comment:
The inclusion of `TVector` here is erroneous. Probably. I guess you could argue about it but really, it doesn't matter. It **will** matter for RAII vectors, but not for legacy vectors.

Reply:
Now no longer delving into `TVector`.